jenkins/compy_mam4xx_compare_performance.sh

```diff
@@ -7,6 +7,7 @@ main() {
     #---------------------------------------------------------------
     #path to E3SM
     code_root="/compyfs/litz372/e3sm_scratch/performance_testing/E3SM"
+    fetch_root=true
```

Comment:
Could you please add comments about this new variable?
"Hard reset code to the most recent master"
Maybe change the name of the variable to `hard_reset_code` to explicitly expose the functionality. It might require changes at some other places as well.